### ISSUE TITLE
`PrimeIdeal` generator should be reduced

### ISSUE THREAD

**Issue by skieffer** [sympy/sympy#23402]
We represent prime ideals by a pair of generators `(p, alpha)`, where `p` is a rational prime,
and `alpha` is an algebraic integer.

It's important that `alpha` be reduced relative to the principal ideal `(p)`, in order to get
an appropriate representation. Currently we are not doing this.

In particular, `alpha` should be exactly `0` if it is equivalent to `0` mod `(p)`.

For example, we currently have this:

```python
>>> from sympy import QQ, Poly, symbols
>>> t = symbols('t')
>>> k = QQ.alg_field_from_poly(Poly(t**3 + t**2 - 2*t + 8))
>>> P = k.primes_above(3)
>>> P[0].alpha.as_expr(t)
3*t - 9
```

In other words, we're representing this prime ideal as `(3, 3*t - 9)`, with a second generator that
contributes nothing. `3*t - 9` should be reduced mod 3, giving 0.
